subsys/pm/pm.c

```diff
@@ -149,6 +149,11 @@ bool pm_system_suspend(int32_t kernel_ticks)
 	int32_t ticks, events_ticks;
 	uint32_t exit_latency_ticks;
 
+	if (pm_cpu_shell_forced_idle()) {
+		/* Directly enter cpu_idle, no about soc level low power mode */
+		return false;
+	}
+
```

Comment:
This does not look correct ... This is just preventing the subsystem to use a power state. You can achieve the same behavior in the right way and without adding any code just getting a lock for all power states.